### ISSUE TITLE
Userdata in getHandlerPairs()?

### ISSUE THREAD

**Issue by ihamburglar** [aws/amazon-ec2-metadata-mock#177]
Does userdata need to be registered, in this function? https://github.com/aws/amazon-ec2-metadata-mock/blob/0965dfc5fbdc2cb574fd304e0e32bb2463e2eb5a/pkg/cmd/cmdutil/cmdutil.go#L102-L109

It seems like things work without it, what does this function do exactly?  Is it actually needed?

**Comment by brycahta** [comment]
> Does userdata need to be registered, in this function?

No, per my comment on your PR: https://github.com/aws/amazon-ec2-metadata-mock/pull/171#issuecomment-1143821018 

> what does this function do exactly? Is it actually needed?

Yup it's needed. It associates a `path` with ` handler` during startup depending on which subcommand is used to start AEMM. For example, if users use events subcommand, `aemm events <args>`,  then IMDS paths (and handlers) involving *spot* should not be available-- that logic is handled in `getHandlerPairs`. 

Addressing this specific block: 

```
	handlerPairs := []handlerPair{
		{path: "/", handler: handlers.ListRoutesHandler},
		{path: "/latest", handler: handlers.ListRoutesHandler},
		{path: static.ServicePath, handler: handlers.ListRoutesHandler},
		{path: dynamic.ServicePath, handler: handlers.ListRoutesHandler},
	}
```

`handlerPairs` is saying each of these paths have a *List Handler*, which is a handler that *lists* available routes for a given path. userdata does **NOT** have a *List Handler* because userdata has no routes itself just the userdata value. Ex:

* `/latest` needs to register a *List Handler* because hitting this path should *list* the following routes:
  * dynamic
  * meta-data
  * user-data

* `/latest/dynamic` needs to register a *List Handler* because hitting this path should *list* the following routes:
  * fws/instance-monitoring
  * instance-identity/document
  * instance-identity/pkcs7
  * instance-identity/signature


* `/latest/user-data` does **not** have a *List Handler* because hitting this path will return **user-data value**:
  *  MTIzNCxqb2huLHJlYm9vdCx0cnVlCg==
